tests/units/txBatch/bulker.test.ts

```diff
@@ -0,0 +1,383 @@
+          ...ADAPTER_MOCK.userData,
+          ethBalance: parseUnits("100"),
+        },
+        userMarketsData: {
+          ...ADAPTER_MOCK.userMarketsData,
+          [Underlying.weth]: {
+            ...ADAPTER_MOCK.userMarketsData[Underlying.weth],
+            walletBalance: parseUnits("50"),
+          },
+        },
+      };
+      const adapter = MorphoAaveV3Adapter.fromMock(mock);
+      bulkerHandler = new BulkerTxHandler(adapter);
+      await adapter.connect(userAddress);
+      await adapter.refreshAll();
+
+      const amount = parseUnits("100");
+      bulkerHandler.addOperation({
+        type: TransactionType.supply,
+        underlyingAddress: Underlying.weth,
+        amount,
+      });
+      const operations = bulkerHandler.getBulkerTransactions();
+      expect(operations).toHaveLength(4);
+
+      expect(operations[0].type).toEqual(Bulker.TransactionType.wrapEth);
+      const wrap = operations[0] as Bulker.WrapEthTransaction;
+      expect(wrap.amount).toEqual(parseUnits("50"));
+
+      expect(operations[1].type).toEqual(Bulker.TransactionType.approve2);
+      const approve = operations[1] as Bulker.Approve2Transaction;
+      expect(approve.amount).toEqual(parseUnits("50"));
+      expect(approve.asset).toEqual(Underlying.weth);
+
+      expect(operations[2].type).toEqual(Bulker.TransactionType.transferFrom2);
+      const transfer = operations[2] as Bulker.TransferFrom2Transaction;
+      expect(transfer.amount).toEqual(parseUnits("50"));
+      expect(transfer.asset).toEqual(Underlying.weth);
+
+      expect(operations[3].type).toEqual(Bulker.TransactionType.supply);
+      const supply = operations[3] as Bulker.SupplyTransaction;
+      expect(supply.amount).toBnEq(amount);
+      expect(supply.asset).toEqual(Underlying.weth);
+
+      expect(bulkerHandler.getValues()).toBnEq(parseUnits("50"));
+    });
+
+    it("should wrap steth", async () => {
+      //  set the  weth balance to 0
+      const mock = {
+        ...ADAPTER_MOCK,
+        userMarketsData: {
+          ...ADAPTER_MOCK.userMarketsData,
+          [Underlying.wsteth]: {
+            ...ADAPTER_MOCK.userMarketsData[Underlying.wsteth],
+            walletBalance: constants.Zero,
+          },
+        },
+      };
+      const adapter = MorphoAaveV3Adapter.fromMock(mock);
+      bulkerHandler = new BulkerTxHandler(adapter);
+      await adapter.connect(userAddress);
+      await adapter.refreshAll();
+
+      const amount = parseUnits("40");
+      bulkerHandler.addOperation({
+        type: TransactionType.supplyCollateral,
+        underlyingAddress: Underlying.wsteth,
+        amount,
+      });
+      const operations = bulkerHandler.getBulkerTransactions();
+      expect(operations).toHaveLength(5);
+
+      expect(operations[0].type).toEqual(Bulker.TransactionType.approve2);
+      const approval = operations[0] as Bulker.Approve2Transaction;
+      expect(approval.amount).toBnEq("40000000000200000000"); // add a buffer for the wrap
```

Comment:
The buffer was supposed to be 1e8 right?

Reply:
that's due to the buffer added twice, you're right